data/mods/ssb/moves.ts

```diff
@@ -96,6 +96,47 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
 		type: "Ice",
 	},
 
+	// Cake
+	kevin: {
+		accuracy: 100,
+		basePower: 111,
+		category: "Physical",
+		desc: "This move combines the user's current typing in its type effectiveness against the target.",
+		shortDesc: "Combines current types in its type effectiveness.",
+		name: "Kevin",
+		pp: 10,
+		flags: {protect: 1, mirror: 1},
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, 'Brave Bird', target);
+		},
+		onModifyMove(move, pokemon, target) {
```

Comment:
we have onModifyType

Reply:
also, I would rather make the first type be the type of the move, and the second type be in the effectiveness.  You dont need to bend over backwards to make the effectiveness for 2 types.